airbyte-integrations/connectors/source-trustpilot/source_trustpilot/streams.py

```diff
@@ -0,0 +1,269 @@
+    ) -> str:
+        return f"business-units/{stream_slice['business_unit_id']}"
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        return {}
+
+    def stream_slices(self, stream_state: Mapping[str, Any] = None, **kwargs) -> Iterable[Optional[Mapping[str, any]]]:
+        all_business_units = _AllBusinessUnitsIterator(authenticator=self._session.auth)
+        for busines_unit_data in all_business_units.read_records(sync_mode=SyncMode.full_refresh):
+            yield {'business_unit_id': busines_unit_data['id']}
+
+
+class PrivateReviews(TrustpilotIncrementalStream):
+    """
+    Business Unit private reviews.
+
+    See also: https://documentation-apidocumentation.trustpilot.com/business-units-api#business-unit-private-reviews
+    """
+    primary_key = 'id'
+    data_field = 'reviews'
+    per_page = 100
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return f"private/business-units/{stream_slice['business_unit_id']}/reviews"
+
+    def request_params(self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None,
+                       next_page_token: Mapping[str, Any] = None) -> MutableMapping[str, Any]:
+        params = super().request_params(stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token)
+        params.update({
+            'orderBy': 'createdat.asc'
+        })
+        return params
+
+    def stream_slices(self, stream_state: Mapping[str, Any] = None, **kwargs) -> Iterable[Optional[Mapping[str, any]]]:
+        """
+        Currently we only support syncing from a specific pre-defined business unit name
+        given in the configuration. Probably in a future version when someone demands
+        it we could add support for generic business units sync.
+        """
+        business_units_find = ConfiguredBusinessUnits(
+            authenticator=TrustpilotApikeyAuthenticator(token=self._api_key),
+            business_unit_names=self._business_unit_names)
+        for stream_slice in business_units_find.stream_slices(sync_mode=SyncMode.full_refresh):
+            for busines_unit_data in business_units_find.read_records(sync_mode=SyncMode.full_refresh, stream_slice=stream_slice):
+                yield {'business_unit_id': busines_unit_data['id']}
+
+    def _clean_row(self, row: Mapping[str, Any]):
+        """
+        A internal function to clean the data from unnecessary data.
+        """
+        row = super()._clean_row(row)
+
+        # remove nested 'links'
+        if 'consumer' in row:
+            if 'links' in row['consumer']:
+                del row['consumer']['links']
+        if 'businessUnit' in row:
+            if 'links' in row['businessUnit']:
```

Comment:
@leo-schick can you explain why are you removing the links? there is any sensitive data in there?

Reply:
The links are RESPA API hypermedia data: They give the receiver a list of links how you can interact with the API. They do not provide meaningful value to the data receiver.

E.g the links for the `consumer` might provide a exact link how to get detailed data about the consumer via the [Consumer API](https://documentation-apidocumentation.trustpilot.com/consumer-api).

In case someone is interested in that, IMO a new `consumer` stream should be added. To save the links is IMO blowing up the data.

p.s. would be great to have a Airbyte helper function to trim REST API hypermedia data.